ggshield/cmd/common_options.py

```diff
@@ -84,24 +84,45 @@ def callback(
 def debug_callback(
     ctx: click.Context, param: click.Parameter, value: Optional[bool]
 ) -> Optional[bool]:
-    # The --debug option is marked as "is_eager" so that we can setup logs as soon as
-    # possible. If we don't then log commands for the creation of the Config instance
-    # are ignored
     if value is not None:
-        setup_debug_logs(value is True)
+        setup_debug_logs(filename=None)
     return value
 
 
+# The --debug option is marked as "is_eager" so that we can setup logs as soon as
+# possible. If we don't then log commands for the creation of the Config instance
+# are ignored.
 _debug_option = click.option(
     "--debug",
     is_flag=True,
     default=None,
     is_eager=True,
-    help="Show debug information.",
+    help="Send log output to stderr. Equivalent to `--log -`.",
     callback=debug_callback,
 )
 
 
+def log_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[str]
+) -> Optional[str]:
+    if value is not None:
+        setup_debug_logs(filename=None if value == "-" else value)
+    return value
+
+
+# The --log option is marked as "is_eager" so that we can setup logs as soon as
+# possible. If we don't then log commands for the creation of the Config instance
+# are ignored.
+_log_option = click.option(
+    "--log",
+    metavar="FILE",
+    is_eager=True,
+    help="Send log output to FILE. Use '-' to redirect to stderr.",
+    envvar="GITGUARDIAN_LOG",
```

Comment:
Click has a feature called [`auto_envvar_prefix`](https://click.palletsprojects.com/en/8.1.x/options/#values-from-environment-variables) which, when enabled, lets users set any CLI option using a matching environment variable.

I did not feel like activating it in this PR, but I picked the name of the environment variable so that if we were to enable it, it would match the name to use to control `--log`. I'd like to keep this mapping.

What do you think of renaming `--log` to `--log-file` (It may actually be more common, for example maven, rsync and valgrind uses it, probably others) and then `GITGUARDIAN_LOG` to `GITGUARDIAN_LOG_FILE`?

Reply:
Makes sense 👍